cranelift/codegen/src/machinst/mod.rs

```diff
@@ -354,49 +354,66 @@ impl<T: CompilePhase> CompiledCodeBase<T> {
 
         let relocs = self.buffer.relocs();
         let traps = self.buffer.traps();
-        let labels = self.bb_starts.as_slice();
 
-        let insns = cs.disasm_all(self.buffer.data(), 0x0).map_err(map_caperr)?;
-        for i in insns.iter() {
-            if let Some((n, off)) = labels
-                .iter()
-                .copied()
-                .enumerate()
-                .find(|(_, val)| *val == i.address() as u32)
-            {
-                writeln!(buf, "block{}: ; offset 0x{:x}", n, off)?;
+        // Normalize the block starts to include an initial block of offset 0.
+        let block_starts = if let Some(start) = self.bb_starts.first() {
+            let mut starts = Vec::new();
+            if *start != 0 {
+                starts.push(0);
             }
+            starts.extend_from_slice(&self.bb_starts);
+            starts
+        } else {
+            vec![0]
+        };
```

Comment:
This makes a little more sense to me:

```suggestion
        let mut block_starts = Vec::new();
        if self.bb_starts.first().copied() != Some(0) {
            block_starts.push(0);
        }
        block_starts.extend_from_slice(&self.bb_starts);
```